blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/NodeRunStatus.java

```diff
@@ -72,8 +73,12 @@ public NodeRunStatus(GenericStatus status){
                 this.state =  BlueRun.BlueRunState.FINISHED;
                 break;
             case NOT_EXECUTED:
-                this.result = BlueRun.BlueRunResult.NOT_BUILT;
-                this.state = BlueRun.BlueRunState.QUEUED;
+                // It happens when declarative plugin strting from 0.5
+                // sends NOT_EXECUTED for nodes thats not run in case earlier node failed.
+                // we want to represent them as totally unknown in blueocean so that UI can
+                // represent them accordingly
+                this.result = null;
```

Comment:
Btw, null was returned so that it doesn't break UI, that is it shows up as a gray circle.